common/src/chain/config/mod.rs

```diff
@@ -637,6 +637,40 @@ pub fn create_unit_test_config() -> ChainConfig {
         .build()
 }
 
+/// This function ensure that IgnoreConsensus will never be used in anything other than regtest
+pub fn assert_no_ignore_consensus_in_chain_config(chain_config: &ChainConfig) {
+    match chain_config.chain_type() {
+        ChainType::Regtest => {
+            return;
+        }
+        ChainType::Mainnet | ChainType::Testnet | ChainType::Signet => {}
+    }
+
+    let upgrades = chain_config.net_upgrade();
+
+    let all_upgrades = upgrades.all_upgrades();
+
+    for upgrade in all_upgrades {
+        let upgrade_height = &upgrade.0;
+        let upgrade_data = &upgrade.1;
+
+        // We skip genesis
+        if *upgrade_height == 0.into() && all_upgrades.len() > 1 {
+            continue;
+        }
```

Comment:
Is the first upgrade required to be at genesis? If so, we can just assert the first upgrade is at height 0 and only iterate over `all_upgrades.into_iter().skip(1)`.

Reply:
Done... despite it being more verbose.